.github/workflows/build.yml

```diff
@@ -0,0 +1,22 @@
+name: "Build"
+on: [push, pull_request]
+
+jobs:
+  validate:
```

Comment:
nit: split to 3 jobs ? 

1. build (make build)
2. lint (make lint)
3. test (make test)

Reply:
I like the idea to make CI faster! I'll do it in a next PR update.